src/routes/ammApy.route.ts

```diff
@@ -62,3 +63,26 @@ router.get(
     }
   })
 )
+
+/**
+ * This endpoint is only for D1, so that the legacy backend can be fully deprecated
+ * It should be deprecated when D1 frontend is deprecated
+ */
+router.get(
+  '/pool-balance/:pool',
+  checkBodyAndParams('pool').exists().isEthereumAddress(),
+  asyncMiddleware(async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      req.log.info('handling pool balance request')
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        req.log.debug(errors.array(), 'handling message errors')
+        throw new InputValidateError(errors.array())
+      }
+      const response = await getPoolBalanceData(req.params.pool)
+      res.status(200).send(response)
+    } catch (error) {
```

Comment:
Reading from https://expressjs.com/en/guide/error-handling.html I don't think it's necessary to catch the error here -- it should automatically propagate. But correct me if I'm wrong -- I haven't used express that much.

Reply:
Interesting. I think you're right, and all the microservices have this implementation. I'll play around with it to check